mpt/impl/imap-mailbox/rabbitmq/src/test/java/org/apache/james/mpt/imapmailbox/rabbitmq/host/RabbitMQEventBusHostSystem.java

```diff
@@ -0,0 +1,197 @@
+
+
+        InMemoryMessageId.Factory messageIdFactory = new InMemoryMessageId.Factory();
+        InMemoryId.Factory mailboxIdFactory = new InMemoryId.Factory();
+        EventSerializer eventSerializer = new EventSerializer(mailboxIdFactory, messageIdFactory);
+        RoutingKeyConverter routingKeyConverter = new RoutingKeyConverter(ImmutableSet.of(new MailboxIdRegistrationKey.Factory(mailboxIdFactory)));
+        rabbitConnectionFactory = createRabbitConnectionFactory();
+        eventBus = new RabbitMQEventBus(rabbitConnectionFactory, eventSerializer, RetryBackoffConfiguration.DEFAULT, routingKeyConverter, new MemoryEventDeadLetters());
+
+        StoreRightManager storeRightManager = new StoreRightManager(mailboxSessionMapperFactory, aclResolver, groupMembershipResolver, eventBus);
+        StoreMailboxAnnotationManager annotationManager = new StoreMailboxAnnotationManager(mailboxSessionMapperFactory, storeRightManager);
+        SessionProvider sessionProvider = new SessionProvider(authenticator, authorizator);
+
+        maxQuotaManager = new InMemoryPerUserMaxQuotaManager();
+        DefaultUserQuotaRootResolver quotaRootResolver =  new DefaultUserQuotaRootResolver(sessionProvider, mailboxSessionMapperFactory);
+        InMemoryCurrentQuotaManager currentQuotaManager = new InMemoryCurrentQuotaManager(new CurrentQuotaCalculator(mailboxSessionMapperFactory, quotaRootResolver), sessionProvider);
+        StoreQuotaManager quotaManager = new StoreQuotaManager(currentQuotaManager, maxQuotaManager);
+        ListeningCurrentQuotaUpdater listeningCurrentQuotaUpdater = new ListeningCurrentQuotaUpdater(currentQuotaManager, quotaRootResolver, eventBus, quotaManager);
+        QuotaComponents quotaComponents = new QuotaComponents(maxQuotaManager, quotaManager, quotaRootResolver, listeningCurrentQuotaUpdater);        MessageSearchIndex index = new SimpleMessageSearchIndex(mailboxSessionMapperFactory, mailboxSessionMapperFactory, new DefaultTextExtractor());
+
+        mailboxManager = new InMemoryMailboxManager(mailboxSessionMapperFactory, sessionProvider, locker, messageParser,
+            messageIdFactory, eventBus, annotationManager,  storeRightManager, quotaComponents, index);
+
+        eventBus.start();
+        eventBus.register(listeningCurrentQuotaUpdater);
+
+        ImapProcessor defaultImapProcessorFactory =
+            DefaultImapProcessorFactory.createDefaultProcessor(
+                mailboxManager,
+                eventBus,
+                sm,
+                quotaComponents.getQuotaManager(),
+                quotaComponents.getQuotaRootResolver(),
+                new DefaultMetricFactory());
+        configure(new DefaultImapDecoderFactory().buildImapDecoder(),
+            new DefaultImapEncoderFactory().buildImapEncoder(),
+            defaultImapProcessorFactory);
+    }
+
+
+    private RabbitMQConnectionFactory createRabbitConnectionFactory() throws URISyntaxException {
```

Comment:
this step really looks alike what RabbitMQExtension does inside it

Reply:
Yes but I'm stuck with junit 4 in MPT

I will push that concern in the Junit rule.